extensions/lifecycle/tasks/LifecycleTaskV2.js

```diff
@@ -236,8 +252,10 @@ class LifecycleTaskV2 extends LifecycleTask {
                         );
                     }
 
-                    return done();
-                });
+                    return resolve();
+                })));
+
+            return Promise.allSettled(promises).then(() => done(), done);
```

Comment:
With this code, in case of error in `_compareRulesToList`, we are ignoring the returned error, as we would have it in the "then" with the `rejected` status. This looks like a behavior change we don't want? We ignore for `_sendBucketEntry` but for `_compareRulesToList`, we actually call the `reject` function...

`Promise.allSettled` will return an array with the status for each promise, including the ones rejected:

```
Promise.allSettled([
  Promise.resolve(33),
  new Promise((resolve) => setTimeout(() => resolve(66), 0)),
  99,
  Promise.reject(new Error("an error")),
]).then((values) => console.log(values), err => console.log('catch', err));

[
    {
        "status": "fulfilled",
        "value": 33
    },
    {
        "status": "fulfilled",
        "value": 66
    },
    {
        "status": "fulfilled",
        "value": 99
    },
    {
        "status": "rejected",
        "reason": {}
    }
]
```

(and no "catch" log).

Reply:
you are right (and I will update the code), but actually `_compareRulesToList` is not upposed to fail either : it calls `_retryEntry`, which ignores the errors. Errors on individual objects are retried a bit ; but eventually we just skip the object (or batch), relying on the periodic scan to eventually process these obejcts.